### ISSUE TITLE
Lose of details

### ISSUE THREAD

**Issue by leonzfa** [gengshan-y/high-res-stereo#7]
Hi,

I used train.py file and the provided final-768px.tar model to finetune SceneFlow dataset. But I got blurry results. The disparity seems to be generated from cost volume of 1/16 resolution. How to produce disparity with details? Thanks.

**Comment by gengshan-y** [comment]
Hi, what is the resolution of the input images? We do predict disparities from 1/16 cost volumes, but the input resolution should be high enough. Can you try changing —testres argument to upscale the inputs to around 5 mega-pixel?

**Comment by leonzfa** [comment]
@gengshan-y I haven't test images yet. I just train the network, and visualize the prediction during training. The max_disp is set to 192. Should I set max_disp to 768 or other value when training?

**Comment by gengshan-y** [comment]
Yes, you are right. At line 68 of train.py, "scale_factor" is set to maxdisp / 384, which in your case becomes 0.5. As a result, the model is fine-tuned with *low-res* SceneFlow images. To solve this, you could directly modify scale_factor as 2.0, or change maxdisp to 768 or 512. 

Also notice the learning rate should be no greater than 1e-4, since it was decayed to 1e-4 during training.

**Comment by hareshnindblr76** [comment]
@gengshan-y I was facing similar issues. When I check the code, the scale_factor is used only on line76 train.py for sceneflow training - rand_scale = (0.9, 2.4*rescale_factor) . Which is later used as the cv2.resize ratio. My question is if we use 192 as maximum disparity, then rand_scale would be 0.9 to 1.2. Why would finetuning occur at low-res SceneFlow? Is it required to upsample Sceneflow images to 1920x1080 to avoid getting blurry results?

**Comment by gengshan-y** [comment]
The blurry issue has almost nothing to do with training. It's an architecture issue. Our model is designed for high-res input images with ~3000x2000 pixels. If you have high-res inputs, that's fine. If there are only low-res inputs, a workaround is to upsample them to high-res, by setting a high value to the —testres argument. Without test-time upsampling, the results of low-res inputs will be blurry. 

Augmentation scaling factors are chosen to match the scale of Middlebury dataset. Not sure if that answers your question.
